Test/VimCoreTest/InsertModeIntegrationTest.cs

```diff
@@ -371,6 +371,108 @@ public void NormalizeLineEndings()
+                Assert.Equal("ë", _textBuffer.GetLine(0).GetText());
+                Assert.Equal(1, _textView.GetCaretPoint().Position);
+            }
+
+            [WpfFact]
+            public void Special()
+            {
+                Create("", "");
+                _vimBuffer.ProcessNotation("<C-k><Right>");
+                Assert.Equal("<Right>", _textBuffer.GetLine(0).GetText());
+                Assert.Equal(7, _textView.GetCaretPoint().Position);
+            }
+
+            [WpfFact]
+            public void Undefined()
+            {
+                Create("", "");
+                _vimBuffer.ProcessNotation("<C-k>AB");
+                Assert.Equal("B", _textBuffer.GetLine(0).GetText());
+                Assert.Equal(1, _textView.GetCaretPoint().Position);
+            }
+
+            [WpfFact]
+            public void Space()
+            {
+                Create("", "");
+                _vimBuffer.ProcessNotation("<C-k><Space>a");
+                Assert.Equal("á", _textBuffer.GetLine(0).GetText());
+                Assert.Equal(1, _textView.GetCaretPoint().Position);
+            }
+
+            [WpfFact]
+            public void Escape1()
+            {
+                Create("", "");
+                _vimBuffer.ProcessNotation("<C-k><Esc>");
+                Assert.Equal("", _textBuffer.GetLine(0).GetText());
+                Assert.Equal(0, _textView.GetCaretPoint().Position);
+                Assert.False(_vimBuffer.InsertMode.IsInPaste);
+            }
+
+            [WpfFact]
+            public void Escape2()
+            {
+                Create("", "");
+                _vimBuffer.ProcessNotation("<C-k>e<Esc>");
+                Assert.Equal("", _textBuffer.GetLine(0).GetText());
+                Assert.Equal(0, _textView.GetCaretPoint().Position);
+                Assert.False(_vimBuffer.InsertMode.IsInPaste);
+            }
+
+            [WpfFact]
+            public void NoInline()
+            {
+                Create("", "");
+                _vimBuffer.ProcessNotation("e<BS>:");
+                Assert.Equal(":", _textBuffer.GetLine(0).GetText());
+                Assert.Equal(1, _textView.GetCaretPoint().Position);
+            }
+
+            [WpfFact]
+            public void Inline()
+            {
+                Create("", "");
+                _globalSettings.Digraph = true;
+                _vimBuffer.ProcessNotation("e<BS>:");
+                Assert.Equal("ë", _textBuffer.GetLine(0).GetText());
+                Assert.Equal(1, _textView.GetCaretPoint().Position);
+            }
+
+            [WpfFact]
+            public void InlineSwapped()
+            {
+                Create("", "");
+                _globalSettings.Digraph = true;
+                _vimBuffer.ProcessNotation(":<BS>e");
+                Assert.Equal("ë", _textBuffer.GetLine(0).GetText());
+                Assert.Equal(1, _textView.GetCaretPoint().Position);
+            }

```

Comment:
Another test to add. Start with the word `CAT` in register and in a blank line in insert mode: `<Ctrl-R>c<BS>H`. That should cause the digraph for TH to be entered. It should excercise a slightly different code path than normal typing.

Reply:
Interesting! Although, vim does "support" that, it seems more a like a unfortunate consequence. Think about the analogous operation (which also works in vim): putting `0` into a register, pasting it and then typing `<C-d>` removes all indent.

Anyway, it does work also in "VsVim" because a text change records a paste as though the user typed it. I can add a test.